<a href="https://colab.research.google.com/github/bookflute/ProgrammingAssignment2/blob/master/question1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import tensorflow as tf
import matplotlib.pyplot as plt
import numpy as np


In [0]:
mnist =tf.keras.datasets.mnist

(x_train, y_train), (x_test,y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0



In [21]:
# model with three hidden layers
model = tf.keras.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
    tf.keras.layers.Dense(128, activation=tf.nn.relu),
    tf.keras.layers.Dense(64, activation=tf.nn.relu),
    tf.keras.layers.Dense(32, activation=tf.nn.relu),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10)
])
model.compile(optimizer = 'adam', 
               loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits = True),
               metrics = ['accuracy'])
model.fit(x_train,y_train,epoach = 5)
# model evaluation 
test_accuracy = model.evaluate(x_test, y_test, verbose = 3)
print('Accurary on test dataset:', test_accuracy)


1875/1875 [==============================] - 4s 2ms/step - loss: 0.3198 - accuracy: 0.9058
Accurary on test dataset: [0.14536866545677185, 0.9553999900817871]


In [22]:
 #model with 32 nodes
 model_32 = tf.keras.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
    tf.keras.layers.Dense(32, activation=tf.nn.relu),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10)
])
model_32.compile(optimizer = 'adam', 
               loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits = True),
               metrics = ['accuracy'])
model_32.fit(x_train,y_train,epoach = 5)
# model evaluation 
test_accuracy = model_32.evaluate(x_test, y_test, verbose = 3)
print('Accurary on test dataset:', test_accuracy)



1875/1875 [==============================] - 3s 1ms/step - loss: 0.4638 - accuracy: 0.8639
Accurary on test dataset: [0.21656596660614014, 0.9358000159263611]


The accuracy of one hiddlen layer is 94% whereas the accuracy of three hidden layers is 96%. 